manimo/actuators/arms/franka_arm.py

```diff
@@ -133,13 +133,39 @@ def _go_home(self):
             self.robot.update_current_policy({"q_desired": joint_position})
             rate.sleep()
 
+    def soft_ctrl(self, action_target, time_to_go=4):
+        goal = torch.Tensor(action_target)
+        
+        # Create policy instance
+        q_initial = self.robot.get_joint_positions()
+        waypoints = toco.planning.generate_joint_space_min_jerk(
+            start=q_initial, goal=goal, time_to_go=time_to_go, hz=self.hz
+        )
+        rate = Rate(self.hz)
+        joint_positions = [waypoint["position"] for waypoint in waypoints]
+
+        q_initial = self.robot.get_joint_positions()
+        kq = torch.Tensor(self.robot.metadata.default_Kq)
+        kqd = torch.Tensor(self.robot.metadata.default_Kqd)
+        policy = JointPDPolicy(
+            desired_joint_pos=q_initial,
+            kq=kq,
+            kqd=kqd,
+        )
+        self.robot.send_torch_policy(policy, blocking=False)
+        rate.sleep()
+        for joint_position in joint_positions:
+            self.robot.update_current_policy({"q_desired": joint_position})
+            rate.sleep()
+            
+        self.connect()
+
```

Comment:
is this necessary

Reply:
yes, this is the soft control with lower control gains, used in reset and some other places